selectrum.el

```diff
@@ -706,30 +706,40 @@ PREDICATE, see `read-buffer'."
          (orig-refine-function selectrum-refine-candidates-function)
          (selectrum-preprocess-candidates-function #'ignore)
          (selectrum-refine-candidates-function
-          (lambda (input _)
-            (let ((candidates (mapcar #'buffer-name (buffer-list))))
-              (if (string-prefix-p " " input)
-                  (progn
-                    (setq input (substring input 1))
-                    (setq candidates
-                          (cl-delete-if-not
-                           (lambda (name)
-                             (string-prefix-p " " name))
-                           candidates)))
-                (setq candidates
-                      (cl-delete-if
-                       (lambda (name)
-                         (string-prefix-p " " name))
-                       candidates)))
-              `((candidates . ,(funcall
-                                orig-refine-function
-                                input
-                                (funcall
-                                 orig-preprocess-function
-                                 candidates)))
-                (input . ,input))))))
-    (selectrum-completing-read
-     prompt nil predicate require-match nil nil def)))
+          (let ((balist ()))
+            ;; predicate gets passed an alist
```

Comment:
I don't think this is necessary; the docstring of `read-buffer` says that the predicate gets passed "each potential candidate, in the form of either a string or a cons cell whose car is a string". So I think we can just remove the preprocessing and do `(when predicate (setq candidates (cl-delete-if-not predicate candidates))` in the `lambda`. Thoughts?

Reply:
My docstring does not say anything about the format (Emacs 26.3). I tested with the default which gets passed an alist so I assumed this. Applying the candidate outside of the lambda still makes sense because it does not need to run on every update I think?